### ISSUE TITLE
[Java] FuryBuilder supports registered classes

### ISSUE THREAD

**Issue by LiangliangSui** [apache/fory#1459]
## Is your feature request related to a problem? Please describe.
Currently, classes can only be registered through the `Fury#register` interface. Could we build Fury and register the class in one go?

Some users have reported that classes cannot be registered through `FuryBuilder`. #1458 

## Describe the solution you'd like
Could we support chained(In the process of building Fury) class registration in `FuryBuilder`?

```java
// e.g. 
Fury fury = Fury.builder().withLanguage(Language.JAVA)
        .requireClassRegistration(true)
        // --------- Chain execution registration.
        .register(Foo.class)
        .register(Foo1.class)
        .build();
```

## Additional context
N / A

**Comment by chaokunyang** [comment]
One concern I have is that BaseFury already supports register classes/serializers, do we still need to add this API to FuryBuilder?

**Comment by LiangliangSui** [comment]
> One concern is that BaseFury already supports register classes/serializers, do we still need to add this API to FuryBuilder?

We can extract all `register(...)` methods in `BaseFury` into an interface, for example named `ClassRegister`, and then let `BaseFury` extends `ClassRegister`, and `FuryBuilder` can also implements `ClassRegister`. `registerSerializer(...)` is similar, we can extract an interface named `SerializerRegister`.

WDYT? @chaokunyang

**Comment by chaokunyang** [comment]
I perfer not. `Fury` is not a `ClassRegister`, there is no such a straightforward `is-a` relationship between these two classes. And it's not that intuitive for a builder class to implement some interface.

**Comment by LiangliangSui** [comment]
I agree with you, `Fury` and `ClassRegister` have a `has-a` relationship.

If we don't implement some interfaces in `FuryBuilder`, then we can only repeatedly define the registerXXX function, or do you have some other good suggestions?

**Comment by chaokunyang** [comment]
I still don't know why users can't invoke `BaseFury` to register a class/serializer?
